include/bipartite/pgr_bipartite_driver.hpp

```diff
@@ -41,96 +41,63 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #include <boost/property_map/vector_property_map.hpp>
 #include <boost/type_traits.hpp>
 #include <boost/graph/bipartite.hpp>
+#include "cpp_common/interruption.h"
 
 
 namespace pgrouting {
     namespace functions {
+
         template<class G>
         class Pgr_Bipartite : public pgrouting::Pgr_messages {
+            
             public:
-
             typedef typename G::V_i V_i;
-            std::vector<pgr_bipartite_rt> print_Bipartite( G &graph)
-            {
+            
+            std::vector<pgr_bipartite_rt> print_Bipartite(
+                    G &graph){
                 std::vector<pgr_bipartite_rt> results;
-                std::vector <boost::default_color_type> partition (graph.num_vertices());
-
-                 
-                auto partition_map = make_iterator_property_map(partition.begin (), boost::get (boost::vertex_index, graph.graph));
-
-                
-                
-                try {
-             // calling the boost function
-                is_bipartite (graph.graph, boost::get (boost::vertex_index, graph.graph), partition_map);
-                     } catch (boost::exception const& ex) {
-                          (void)ex;
-                             throw;
-                     } catch (std::exception &e) {
-                         (void)e;
-                         throw;
-                      } catch (...) {
-                          throw;
-                      }
-              //  
-                
-                
-                
-            V_i v, vend;
-
-         // iterate through every vertex in the graph
+                std::vector <boost::default_color_type> partition (graph.num_vertices());    
+                auto partition_map =
+                    make_iterator_property_map(partition.begin (), boost::get (boost::vertex_index, graph.graph));
+
+               /* abort in case of an interruption occurs (e.g. the query is being cancelled) */
+                CHECK_FOR_INTERRUPTS();
+                boost::is_bipartite (graph.graph, boost::get (boost::vertex_index, graph.graph), partition_map);
```

Comment:
You removed the try/catch please keep it

Reply:
okay ma'am